pkg/apiserver/registry/stats/nodelatencystats/rest.go

```diff
@@ -50,38 +57,80 @@ func (r *REST) Destroy() {
+	summary := obj.(*statsv1alpha1.NodeLatencyStats)
+	err := r.indexer.Update(summary)
+	if err != nil {
+		return nil, errors.NewInternalError(err)
+	}
+
+	return summary, nil
 }
 
 func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
-	// TODO: fill this function in next PR
-	return &statsv1alpha1.NodeLatencyStats{}, nil
+	obj, exists, err := r.indexer.GetByKey(name)
+	if err != nil || !exists {
+		return nil, errors.NewNotFound(statsv1alpha1.Resource("nodelatencystats"), name)
+	}
+
+	return obj.(*statsv1alpha1.NodeLatencyStats), nil
 }
 
 func (r *REST) NewList() runtime.Object {
 	return &statsv1alpha1.NodeLatencyStats{}
 }
 
 func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
-	// TODO: fill this function in next PR
-	return &statsv1alpha1.NodeLatencyStatsList{}, nil
+	objs := r.indexer.List()
+	if objs == nil {
+		return &statsv1alpha1.NodeLatencyStatsList{}, nil
+	}
+
+	entries := make([]statsv1alpha1.NodeLatencyStats, 0, len(objs))
+	for _, obj := range objs {
+		entries = append(entries, *obj.(*statsv1alpha1.NodeLatencyStats))
+	}
+
+	return &statsv1alpha1.NodeLatencyStatsList{Items: entries}, nil
 }
 
 func (r *REST) ConvertToTable(ctx context.Context, obj runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
-	// TODO: fill this function in next PR
 	table := &metav1.Table{
 		ColumnDefinitions: []metav1.TableColumnDefinition{
 			{Name: "SourceNodeName", Type: "string", Format: "name", Description: "Source node name."},
-			{Name: "NodeLatencyStatsList", Type: "array", Format: "string", Description: "Node IP latency list."},
+			{Name: "PeerNodeLatencyStats", Type: "array", Format: "string", Description: "Current Node to each peers latency."},
 		},
 	}
+	if m, err := meta.ListAccessor(obj); err == nil {
+		table.ResourceVersion = m.GetResourceVersion()
+		table.Continue = m.GetContinue()
+		table.RemainingItemCount = m.GetRemainingItemCount()
+	} else {
+		if m, err := meta.CommonAccessor(obj); err == nil {
+			table.ResourceVersion = m.GetResourceVersion()
+		}
+	}
 
-	return table, nil
+	var err error
+	table.Rows, err = metatable.MetaToTableRow(obj, func(obj runtime.Object, m metav1.Object, name, age string) ([]interface{}, error) {
+		summary := obj.(*statsv1alpha1.NodeLatencyStats)
+		return []interface{}{name, summary.PeerNodeLatencyStats}, nil
+	})
+	return table, err
 }
 
 func (r *REST) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
-	return &statsv1alpha1.NodeLatencyStats{}, true, nil
+	// Ignore the deleteValidation and options for now.
+	obj, exists, err := r.indexer.GetByKey(name)
+	if err != nil || !exists {
+		return nil, false, errors.NewNotFound(statsv1alpha1.Resource("nodelatencystats"), name)
+	}
+
+	err = r.indexer.Delete(obj)
+	if err != nil {
+		return nil, false, err
```

Comment:
do you need to wrap this with `errors.NewInternalError` to be consistent with other methods?

Reply:
Ok I miss it, I will add it.